test/test-helpers.js

```diff
@@ -112,12 +112,21 @@ let fakeEnvironment = {
     this.clock.restore();
     videojs.xhr.XMLHttpRequest = window.XMLHttpRequest;
     this.xhr.restore();
+    ['warn', 'error'].forEach((level) => {
+      if (videojs.log && videojs.log[level] && videojs.log[level].restore) {
+        videojs.log[level].restore();
+      }
+    });
+
   }
 };
 
 export const useFakeEnvironment = function() {
   fakeEnvironment.clock = sinon.useFakeTimers();
-
+  ['warn', 'error'].forEach((level) => {
+    sinon.stub(videojs.log, level);
```

Comment:
I think we may be able to get rid of this issue without silencing the log, took a few minutes to narrow it down, I will write a test around it to verify that nothing changes basically it involves the depreciation warning when we use `player.hls`. when we were checking if `player.hls` existed when it did exist we inadvertently called it and caused the warning to showup.